10-autoencoder.ipynb

```diff
@@ -146,37 +177,53 @@
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 532
+    },
+    "colab_type": "code",
+    "id": "rhiHWX5j2Gq5",
+    "outputId": "7814a936-1709-4b4d-e9c6-559493d4824a",
     "scrolled": false
    },
    "outputs": [],
    "source": [
     "# Train standard or denoising autoencoder (AE)\n",
     "\n",
     "num_epochs = 1\n",
-    "# do = nn.Dropout()  # comment out for standard AE\n",
+    "do = nn.Dropout()  # comment out for standard AE\n",
     "for epoch in range(num_epochs):\n",
     "    for data in dataloader:\n",
     "        img, _ = data\n",
     "        img = img.to(device)\n",
     "        img.requires_grad_()\n",
     "        img = img.view(img.size(0), -1)\n",
-    "#         img_bad = do(img).to(device)  # comment out for standard AE\n",
```

Comment:
This is where the noise is introduced, by sending `img` to `do()` which is a dropout module.